File: java/ecbenchmark/src/ecbenchmark/cec10/CecRotated.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package ecbenchmark.cec10;

import ecbenchmark.Operator;
import ecbenchmark.Function;

/**
 *
 * @author jcrada
 */
public class CecRotated extends Operator {

    protected double[][] rotation;

    public CecRotated(double[][] rotation, Function f) {
        super(f);
        this.rotation = rotation;
    }

    public double f(final double[] x) {
        double[] rotated_x = CecMath.Multiply(x, rotation);
        return innerFunction.f(rotated_x);
    }
}

File: cpp/ecbenchmark/src/cec10/CecRotated.cpp
#include "ecbenchmark/cec10/CecRotated.h"
#include "ecbenchmark/cec10/CecMath.h"

namespace ecb {

    namespace cec10 {

        CecRotated::CecRotated(const std::vector<std::vector<scalar> >& rotation, Function* f)
        : Operator(f), _rotation(rotation) {

        }

        void CecRotated::setRotation(const std::vector<std::vector<scalar> >& rotation) {
            this->_rotation = rotation;
        }

        std::vector<std::vector<scalar> > CecRotated::rotation() const {
            return this->_rotation;
        }

        scalar CecRotated::f(const std::vector<scalar>& x) {
            std::vector<scalar> rotated_x = CecMath::Multiply(x, _rotation);
            return _innerFunction->f(rotated_x);
        }
    }
}
